### PR TITLE
Added AutoRespawn to DeathScreen

### LINKED ISSUE #717: AutoRespawn toggle button on the death screen

### PR DESCRIPTION
**Pull request by TheWienerMaster**
## Description
Resolves #717

## Demonstration 

https://user-images.githubusercontent.com/61437644/222935348-7c1913bb-c5cd-4a78-8ba9-772cd46bcccd.mp4

### CODE REVIEW

**Comment by Alexander01998** [comment]
Still relevant.

**Comment by TheWienerMaster** [comment]
.

File: src/main/java/net/wurstclient/mixin/DeathScreenMixin.java [hunk 1]
```diff
@@ -27,9 +29,28 @@ private DeathScreenMixin(WurstClient wurst, Text title)
...
+			return;
+		
+		int backButtonX = width / 2 - 100;
+		int backButtonY = height / 4;
+		
+		addDrawableChild(
+			ButtonWidget.builder(Text.literal("AutoRespawn: OFF"), b -> {
+				autoRespawn.setEnabled(true);
+				autoRespawn.onDeath();
+			}).dimensions(backButtonX, backButtonY + 48, 200, 20).build());
```
[the remark this answers is not in the document]
Comment (Alexander01998) [reply src/main/java/net/wurstclient/mixin/DeathScreenMixin.java:54]: Nope. `shouldShowButton()` only returns `true` when AutoRespawn is off. @nobotreply

File: src/main/java/net/wurstclient/mixin/DeathScreenMixin.java [hunk 2]
```diff
@@ -27,9 +29,28 @@ private DeathScreenMixin(WurstClient wurst, Text title)
...
+	}
```
[the remark this answers is not in the document]
Comment (Alexander01998) [reply src/main/java/net/wurstclient/mixin/DeathScreenMixin.java:55]: `HackList.autoRespawnHack` is a final field, meaning it can't be null. @nobotreply

**Pull request merged**